articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,803 @@
+---
+layout: default
+title: ROS 2 Launch XML Format
+permalink: articles/roslaunch_xml.html
+abstract:
+  The XML format for declarative launch descriptions in the ROS 2 launch system.
+author: '[Michel Hidalgo](https://github.com/hidmic)'
+published: false
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Authors: {{ page.author }}
+
+
+# ROS 2 Launch XML Format v0.1.0
+
+## Rationale
+
+As an alternative to a programmatic approach to the ROS 2 launch system's API, a declarative description features a WYSIWYG approach, easier to read, audit and maintain.
+This is the preferred approach for ROS 1 `roslaunch` launch files, thus some degree of familiarity is expected (and relied upon).
+
+The body of such a description is mainly comprised of statically declared launch actions with a prescribed configuration.
+To that, runtime value substitution is added in order to fullfill common dynamic (re)configuration needs like conditional execution, resource lookups, etc.
+It is intended for these entities to map to those of the underlying implementation, reducing support to file parsing.
+
+The choice of XML over other markup languages aims to ease the bridge between ROS and ROS 2 launch files.
+Widely available support for parsing XML in a myriad of languages and platforms also weighed (significantly) in the decision.
+
+## Static Description
+
+### Schema Definition
+
+```xml
+<?xml version="1.0"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+  <xs:annotation>
+    <xs:documentation xml:lang="en">
+      ROS 2 launch XML schema v0.1.0
```

Comment:
I think we may want to require a "version" attribute in the root element of the DOM so that it is easier for launcher applications or a version adapter service to figure out what it is processing. E.g:

``` xml
<launch version="0.1.0">
...
```

https://stackoverflow.com/questions/2014237/what-are-the-best-practices-for-versioning-xml-schemas

Reply:
That makes a lot of sense. Added `version` to `<launch>` tag.